### PR TITLE
Fix distance in checkHighPriorityChunks (Fixes #4582)

### LINKED ISSUE #4582: check high priority chunks (teleport)
Hello!
It looks like the root is drawn twice from a distance.

PlayerChunkMap#checkHighPriorityChunks
```
...
        MCUtil.getSpiralOutChunks(pos, Math.min(6, viewDistance)).forEach(coord -> {
            if (shouldSkipPrioritization(coord)) return;

            double dist = MCUtil.distance(playerChunkX, 0, playerChunkZ, coord.x, 0, coord.z);
            // Prioritize immediate
            if (dist <= 4 * 4) {
                updateChunkPriorityMap(priorities, coord.pair(), (int) (27 - Math.sqrt(dist)));
                return;
            }

            // Prioritize nearby chunks
            updateChunkPriorityMap(priorities, coord.pair(), (int) (20 - Math.sqrt(dist) * twoThirdModifier));
        });

..
```

MCUtil#distance
```
    public static double distance(double x1, double y1, double z1, double x2, double y2, double z2) {
        return Math.sqrt(distanceSq(x1, y1, z1, x2, y2, z2));
    }
```

### PR DESCRIPTION
**Pull request by Foorcee**
Fixes #4582

### CODE REVIEW

File: Spigot-Server-Patches/0488-Implement-Chunk-Priority-Urgency-System-for-Chunks.patch [hunk 1]
```diff
@@ -1050,7 +1050,7 @@ index fba7c9b5724114eab35a3d24febdee3cd3e30aa0..753d67c8a711d293bd70eb913c47ec81
 +            MCUtil.getSpiralOutChunks(pos, 4).forEach(coord -> {
 +                if (shouldSkipPrioritization(coord)) return;
 +
-+                double dist = MCUtil.distance(playerChunkX, 0, playerChunkZ, coord.x, 0, coord.z);
++                double dist = MCUtil.distanceSq(playerChunkX, 0, playerChunkZ, coord.x, 0, coord.z);
 +                updateChunkPriorityMap(priorities, coord.pair(), (int) (25 - Math.sqrt(dist) * twoThirdModifier));
```
Comment (kennytv) [inline Spigot-Server-Patches/0488-Implement-Chunk-Priority-Urgency-System-for-Chunks.patch]:
```suggestion
+                double dist = MCUtil.distance(playerChunkX, 0, playerChunkZ, coord.x, 0, coord.z);
+                updateChunkPriorityMap(priorities, coord.pair(), (int) (25 - dist * twoThirdModifier));
```

File: Spigot-Server-Patches/0488-Implement-Chunk-Priority-Urgency-System-for-Chunks.patch [hunk 2]
```diff
@@ -1063,7 +1063,7 @@ index fba7c9b5724114eab35a3d24febdee3cd3e30aa0..753d67c8a711d293bd70eb913c47ec81
 +                if (shouldSkipPrioritization(coord)) {
 +                    return;
 +                }
-+                double dist = MCUtil.distance(playerChunkX, 0, playerChunkZ, coord.x, 0, coord.z);
++                double dist = MCUtil.distanceSq(playerChunkX, 0, playerChunkZ, coord.x, 0, coord.z);
 +                updateChunkPriorityMap(priorities, coord.pair(), (int) (25 - Math.sqrt(dist) * twoThirdModifier));
```
Comment (kennytv) [inline Spigot-Server-Patches/0488-Implement-Chunk-Priority-Urgency-System-for-Chunks.patch]:
```suggestion
+                double dist = MCUtil.distance(playerChunkX, 0, playerChunkZ, coord.x, 0, coord.z);
+                updateChunkPriorityMap(priorities, coord.pair(), (int) (25 - dist * twoThirdModifier));
```

File: Spigot-Server-Patches/0488-Implement-Chunk-Priority-Urgency-System-for-Chunks.patch [hunk 3]
```diff
@@ -1039,7 +1039,7 @@ index fba7c9b5724114eab35a3d24febdee3cd3e30aa0..753d67c8a711d293bd70eb913c47ec81
 +        MCUtil.getSpiralOutChunks(pos, Math.min(5, viewDistance)).forEach(coord -> {
 +            if (shouldSkipPrioritization(coord)) return;
 +
-+            double dist = MCUtil.distance(playerChunkX, 0, playerChunkZ, coord.x, 0, coord.z);
++            double dist = MCUtil.distanceSq(playerChunkX, 0, playerChunkZ, coord.x, 0, coord.z);
 +            updateChunkPriorityMap(priorities, coord.pair(), (int) (25 - Math.sqrt(dist) * twoThirdModifier));
```
Comment (kennytv) [inline Spigot-Server-Patches/0488-Implement-Chunk-Priority-Urgency-System-for-Chunks.patch]:
```suggestion
+            double dist = MCUtil.distance(playerChunkX, 0, playerChunkZ, coord.x, 0, coord.z);
+            updateChunkPriorityMap(priorities, coord.pair(), (int) (25 - dist * twoThirdModifier));
```

**Review by kennytv — CHANGES_REQUESTED** [review]

**Review by electronicboy — DISMISSED** [review]

**Review by aikar — APPROVED** [review]
ignore the comment about I thought it was wrong. Initially I had used squared results to avoid some operations and thought I was double square rooting intentionally in some of those places, but after further review it was good as is as my impl changed over time for this code and I guess that's where things got mixed up.

**Pull request merged**